### PR TITLE
feat: migrate entities to electrodb

### PR DESCRIPTION
**Pull request by solaris007**
This PR significantly enhances the framework built on top of ElectroDB introduced via https://github.com/adobe/spacecat-shared/pull/447, providing a standardized approach to defining entities, schemas, models, and collections. It enforces schema consistency, automatically generates query methods, and supports relationship management—including cascading removals using transactional deletes.

In addition, this PR migrates all currently available schemas, models, and collections from our legacy data layer into this new framework.

### CODE REVIEW

File: packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+
+    const record = await this.entity.get({ [this.idName]: id }).go();
+
+    return this.#createInstance(record?.data);
+  }
+
+  /**
+   * Finds a single entity by index keys.
+   * @param {Object} keys - The index keys to use for the query.
+   * @param {Object} options - Additional options for the query.
```
Comment (dzehnder) [inline packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js:334]: Please provide examples in the JSDoc for the additional options like above

File: packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+
+      if (validatedItems.length > 0) {
+        const response = await this.entity.put(validatedItems).go();
+
+        if (Array.isArray(response?.unprocessed) && response?.unprocessed?.length > 0) {
+          this.log.error(`Failed to process all items in batch write for [${this.entityName}]: ${JSON.stringify(response.unprocessed)}`);
+        }
+      }
+
+      const createdItems = this.#createInstances(validatedItems);
```
Comment (dzehnder) [inline packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js:422]: If a validated item fails to be added to the database using the `put` command,  it still creates an instance and returns the item as successful like this. 
I would expect the method to return the failed/unprocessed items in the `errorItems` instead

File: packages/spacecat-shared-data-access/src/v2/models/import-url/import-url.collection.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * @class ImportUrlCollection
+ * @extends BaseCollection
+ */
+class ImportUrlCollection extends BaseCollection {
+  /**
+   * Constructs an instance of ImportUrlCollection. Tells the base class which model to use.
+   * @constructor
+   * @param {Object} service - The ElectroDB service instance used to manage ImportUrl entities.
+   * @param {Object} entityRegistry - The registry holding entities, their schema and collection..
```
Comment (blefebvre) [inline packages/spacecat-shared-data-access/src/v2/models/import-url/import-url.collection.js:28]:
```suggestion
   * @param {Object} entityRegistry - The registry holding entities, their schema and collection.
```

**Review by blefebvre — COMMENTED** [review]
I've reviewed the import-[job|url] related changes — all appear to be in order. Few comments left, more for my understanding.

I've asked @swetabar to kindly review the api-key related changes.

File: packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+      throw error;
+    }
+  }
+
+  /**
+   * Updates a collection of entities in the database using a batch write (put) operation.
+   *
+   * @async
+   * @param {Array<BaseModel>} items - An array of model instances to be updated.
+   * @return {Promise<void>} - A promise that resolves when the update operation is complete.
```
[the remark this answers is not in the document]
Comment (solaris007) [reply packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js:443]: i'd like to table this for now, it only happens if throughput capacity is exceeded.

File: packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js (hunk 2 above)
Comment (solaris007) [reply packages/spacecat-shared-data-access/src/v2/models/base/base.collection.js:422]: i'd like to table this for now, it only happens if throughput capacity is exceeded.

File: packages/spacecat-shared-data-access/src/v2/models/import-job/import-job.collection.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+   * @constructor
+   * @param {Object} service - The ElectroDB service instance used to manage ImportJob entities.
+   * @param {Object} entityRegistry - The registry holding entities, their schema and collection..
+   * @param {Object} log - A logger for capturing logging information.
+   */
+  constructor(service, entityRegistry, log) {
+    super(service, entityRegistry, ImportJob, log);
+  }
+
+  // add custom methods here
```
[the remark this answers is not in the document]
Comment (solaris007) [reply packages/spacecat-shared-data-access/src/v2/models/import-job/import-job.collection.js:38]: yes, as a hint to developers that they can customize a model

File: packages/spacecat-shared-data-access/src/v2/models/import-job/import-job.collection.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      throw new ValidationError(`Invalid start date: ${startDate}`);
+    }
+
+    if (!isIsoDate(endDate)) {
+      throw new ValidationError(`Invalid end date: ${endDate}`);
+    }
+
+    return this.all({}, {
+      between: {
+        attribute: 'startedAt',
```
[the remark this answers is not in the document]
Comment (solaris007) [reply packages/spacecat-shared-data-access/src/v2/models/import-job/import-job.collection.js:51]: yes, the changes are an attempt to standardize. if you're ok with this, we can still have the api-service DTO map to the old name.

File: packages/spacecat-shared-data-access/src/v2/models/import-job/import-job.schema.js [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    type: 'string',
+    required: true,
+    validate: (value) => isValidUrl(value),
+  })
+  .addAttribute('duration', {
+    type: 'number',
+    default: 0,
+    validate: (value) => !value || isNumber(value),
+  })
+  .addAttribute('endedAt', {
```
[the remark this answers is not in the document]
Comment (solaris007) [reply packages/spacecat-shared-data-access/src/v2/models/import-job/import-job.schema.js:81]: yes, the changes are an attempt to standardize. if you're ok with this, we can still have the api-service DTO map to the old name.

**Review by dzehnder — APPROVED** [review]
LGTM

**Pull request merged**